### PR TITLE
[Spec] Defaults specs environment to Tests

### PR DESCRIPTION
**Pull request by eliasjpr**
Issue: https://github.com/amberframework/amber/issues/491

### Description of the Change

Defaults spec to test environment for an Amber project.

### CODE REVIEW

File: src/amber/cli/templates/app/spec/spec_helper.cr [hunk 1]
```diff
@@ -1,3 +1,5 @@
+ENV["AMBER_ENV"] ||= "test"
```
Comment (veelenga) [inline src/amber/cli/templates/app/spec/spec_helper.cr]: Why is this set conditionally? User can accidentally export `AMBER_ENV`, forget about it and then run specs. ie:

```crystal
$ export AMBER_ENV=production
# do some stuff...
$ crystal spec
```

Comment (eliasjpr) [reply src/amber/cli/templates/app/spec/spec_helper.cr]: The reason is that you can have the AMBER_ENV can be different for CI environment

Comment (robacarp) [reply src/amber/cli/templates/app/spec/spec_helper.cr]: Would it be sufficient to exit when `AMBER_ENV == production`? Or perhaps issue a compile-time warning when `AMBER_ENV` already exists?

```crystal
{% if ENV["AMBER_ENV"]
puts "Warning: AMBER_ENV is set to #{ENV["AMBER_ENV"]}"
end %}
```

Comment (eliasjpr) [reply src/amber/cli/templates/app/spec/spec_helper.cr]: @veelenga what I am trying to state is that having AMBER_ENV set to `test` will remove the ability from devs to use a different `environment.yml` config file for a continuous integration environment like travis.

See https://github.com/rspec/rspec-rails/blob/master/lib/generators/rspec/install/templates/spec/rails_helper.rb#L3

Comment (veelenga) [reply src/amber/cli/templates/app/spec/spec_helper.cr]: I got. But look two lines below:

https://github.com/rspec/rspec-rails/blob/master/lib/generators/rspec/install/templates/spec/rails_helper.rb#L5-L6

You may do the same here.

Comment (eliasjpr) [reply src/amber/cli/templates/app/spec/spec_helper.cr]: Nice catch!

**Pull request merged**